### DISCUSSION TITLE
Navigation sidebar should retain scroll position when opening a new topic

### DISCUSSION THREAD

**Discussion by samsav** [pradyunsg/furo#679] [Ideas]
When the left sidebar has more than one screen of navigation content such that the user needs to scroll the sidebar to see the whole content hierarchy, clicking a topic that's below the first screen causes the sidebar to reset to the top instead of retaining the position to which the user scrolled it.

![furo-navigation-sidebar-jumps-to-top_resized2-repeat3](https://github.com/pradyunsg/furo/assets/6501597/c0a06e5c-ff67-45a1-9c91-5f77cc8ed2fa)

This makes navigating the docset more difficult, since the user needs to scroll the sidebar view again to see the section or topic they navigated to. You can try this out at https://docs.kanzi.com/3.9.6/en/

Ideally, the sidebar would retain its scroll position when loading a new page. I know that Furo is biased towards smaller documentation sets where this issue is less likely to appear, but has this sort of behavior for the sidebar been considered, or could it be considered for future development?

**Comment by pradyunsg** [comment]
I agree that this would be nice. This was something that initial version of Furo tried to do, but doing it in a cross-browser manner that didn't mess with the page-level scrollbar was non-trivial.

I don't think we can preserve scroll position across page loads as requested though -- I think it's clearer to ensure that the current page is visible in the sidebar.

**Reply by samsav** [reply]
Got it! Ensuring that the current page is visible would be very helpful too. 👍

**Reply by lognaturel** [reply]
Making sure that the current page is visible would be a huge help, agreed. Would you be open to a community contribution fo rthis, @pradyunsg?

**Comment by b10902118** [comment]
Any fix for this? Other themes handles this with session storage:
https://github.com/pydata/pydata-sphinx-theme/blob/a00929389cdcd603f7fe1c79200f2b93abf33089/src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js#L145

I can submit PR if this is a desired solution.
